### PR TITLE
[WIP] Added omml conversion methods for formula to OMML conversion

### LINKED ISSUE #28: Implement Formula to OMML conversion
The plurimath gem should re-implement the functionality of converting a Formula into OMML:

Please refer to the code in the [omml2mathml gem](https://github.com/plurimath/omml2mathml)

### PR DESCRIPTION
**Pull request by suleman-uzair**
closes #28  

Added _```to_omml```_ method for **Formula** to **OMML** conversion(reverse conversion).

### CODE REVIEW

File: lib/plurimath/math/formula.rb [hunk 1]
```diff
@@ -43,6 +43,78 @@ def to_html
...
+        if value.length == 2 && ["underover", "powerbase"].include?(
+          value.first.class_name,
+        )
+          nary_tag
+        else
+          <<~OMML
+            <m:r>
+              #{rpr_tag if ['symbol', 'number', 'text'].include?(value.first.class_name)}
+              #{omml_content}
+            </m:r>
```
Comment (ronaldtse) [inline lib/plurimath/math/formula.rb:94]: In Ox, this would be embedding a node within this outer node.

**Review by ronaldtse — CHANGES_REQUESTED** [review]
We want to replace all XML node generation/embedding with Ox functionality

**Review by ronaldtse — APPROVED** [review]
Much better, thanks @suleman-uzair !

File: lib/plurimath/math/formula.rb [hunk 2]
```diff
@@ -43,6 +43,70 @@ def to_html
...
+          "xmlns:wne": "http://schemas.microsoft.com/office/word/2006/wordml",
+          "xmlns:wp": "http://schemas.openxmlformats.org/drawingml/2006/wordprocessingDrawing",
+          "xmlns:wp14": "http://schemas.microsoft.com/office/word/2010/wordprocessingDrawing",
+          "xmlns:wpc": "http://schemas.microsoft.com/office/word/2010/wordprocessingCanvas",
+          "xmlns:wpg": "http://schemas.microsoft.com/office/word/2010/wordprocessingGroup",
+          "xmlns:wpi": "http://schemas.microsoft.com/office/word/2010/wordprocessingInk",
+          "xmlns:wps": "http://schemas.microsoft.com/office/word/2010/wordprocessingShape",
+        }
+        para_element = Utility.omml_element("m:oMathPara", attributes)
+        math_element = Utility.omml_element("m:oMath")
```
Comment (ronaldtse) [inline lib/plurimath/math/formula.rb:68]: Don't you need `attributes`?

File: lib/plurimath/math/formula.rb [hunk 3]
```diff
@@ -43,6 +43,70 @@ def to_html
...
+        Utility.update_nodes(math_element, omml_content)
+        para_element << math_element
+        Ox.dump(para_element)
+      end
+
+      def omml_content
+        value.map do |object|
+          if object.is_a?(Symbol)
+            mt = Utility.omml_element("m:t")
```
Comment (ronaldtse) [inline lib/plurimath/math/formula.rb:77]: No need for attributes?

File: lib/plurimath/math/formula.rb (hunk 2 above)
Comment (ronaldtse) [reply lib/plurimath/math/formula.rb:68]: If you're calling Utility.omml_element with attributes every time, perhaps you should have a builder class that just supplies the `attributes` initially, e.g.

```ruby
builder = Utility::Builder.new(type: :omml, attributes: attributes)
builder.element('m:oMathPara')
# or, builder.element('oMathPara', namespace: 'm')
```

Comment (suleman-uzair) [reply lib/plurimath/math/formula.rb:68]: > If you're calling Utility.omml_element with attributes every time, perhaps you should have a builder class that just supplies the `attributes` initially, e.g.
> 
> ```ruby
> builder = Utility::Builder.new(type: :omml, attributes: attributes)
> para_element = builder.element('m:oMathPara')
> # or, builder.element('oMathPara', namespace: 'm')
> ```

We are not passing attributes for every element(OMML tag).
The ones that are passed contains different attributes for each element.
Not sure how builder class can help in this case. Can you explain the flow for builder class?

File: lib/plurimath/math/formula.rb (hunk 3 above)
Comment (suleman-uzair) [reply lib/plurimath/math/formula.rb:77]: > No need for attributes?

No need of attributes here.

File: lib/plurimath/math/formula.rb (hunk 2 above)
Comment (ronaldtse) [reply lib/plurimath/math/formula.rb:68]: Ah okay

File: lib/plurimath/math/formula.rb (hunk 3 above)
Comment (ronaldtse) [reply lib/plurimath/math/formula.rb:77]: OK.

**Pull request merged**